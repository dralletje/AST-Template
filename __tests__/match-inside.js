let { template, match_inside } = require('../ast-template.js');

it('should match a react component', () => {
  let source = `
    <Route path="/asdf" />
  `;

  let t = template.statements`
    <Route path={${template.Expression('hey')}} />
  `;

  expect(t.match(source)).toMatchSnapshot();
})

it('should match a react component', () => {
  let source = `
    <Route path="/asdf" />
  `;

  let t = template.statements`
    <Route ${template.JSXIdentifier('z')}={${template.Expression('hey')}} />
  `;

  expect(t.match(source)).toMatchSnapshot();
})

it('should match a shorthand jsx attribute', () => {
  let source = `
    <Route exact />
  `;

  let t = template.statements`
    <Route exact />
  `;

  expect(t.match(source)).toMatchSnapshot();
})

it.skip('should match a shorthand jsx attribute to expanded version', () => {
  let source = `
    <Route exact />
  `;

  let t = template.statements`
    <Route exact={true} />
  `;

  console.log(`t.match(source):`, t.match(source))
});

it('should match multiple jsx attributes', () => {
  let source = `
    <Route path="/asdf" />
  `;

  let t = template.statements`
    <Route ${template.many('y', template.jsxAttribute`${template.JSXIdentifier('l')}={${template.Expression('hey')}}`)} />
  `;

  expect(t.match(source)).toMatchSnapshot();
})

it.only('should match multiple boolean attributes', () => {
  let source = `
    <Route exact />
  `;

  let t = template.statements`
    <Route
      ${template.JSXIdentifier('l')}
    />
  `;

  expect(t.match(source)).toMatchSnapshot();
})

it('should find React Components inside a file', () => {
  let source = `
    let App = () => {
      return (
        <div>
          <Route path="/asdf" x={10 } />
          <Route path="/heythere" x={<Route path="/heythere" x={12} />} />
        </div>
      )
    }
  `;

  let xs = match_inside(source, template.expression`
    <Route path={${template.Expression('path')}} x={${template.Expression('x')}} />
  `);

  expect(xs).toMatchSnapshot();
})
