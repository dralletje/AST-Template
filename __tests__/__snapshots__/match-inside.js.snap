// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find React Components inside a file 1`] = `
Array [
  Object {
    "areas": Object {
      "path": "\\"/asdf\\"",
      "x": "10",
    },
  },
  Object {
    "areas": Object {
      "path": "\\"/heythere\\"",
      "x": "<Route path=\\"/heythere\\" x={12} />",
    },
  },
]
`;

exports[`should match a react component 1`] = `
Object {
  "areas": Object {
    "hey": "\\"/asdf\\"",
  },
}
`;

exports[`should match a react component 2`] = `
Object {
  "areas": Object {
    "hey": "\\"/asdf\\"",
    "z": "path",
  },
}
`;

exports[`should match a shorthand jsx attribute 1`] = `
Object {
  "areas": Object {},
}
`;

exports[`should match multiple boolean attributes 1`] = `
Object {
  "areas": Object {
    "l": "exact",
  },
}
`;

exports[`should match multiple jsx attributes 1`] = `
Object {
  "areas": Object {
    "y": Array [
      Object {
        "hey": "\\"/asdf\\"",
        "l": "path",
      },
    ],
  },
}
`;
